daft/sql/sql_scan.py

```diff
@@ -142,22 +140,23 @@ def _attempt_partition_bounds_read(self, num_scan_tasks: int) -> tuple[Any, Part
         try:
             # Try to get percentiles using percentile_cont
             percentiles = [i / num_scan_tasks for i in range(num_scan_tasks + 1)]
-            pa_table = self.conn.read(
-                self.sql,
+            sql = self._construct_sql_query(
                 projection=[
-                    f"percentile_cont({percentile}) WITHIN GROUP (ORDER BY {self._partition_col}) AS bound_{i}"
+                    f"percentile_disc({percentile}) WITHIN GROUP (ORDER BY {self._partition_col}) AS bound_{i}"
```

Comment:
Small note here that I changed this to be discrete percentiles vs continuous percentiles. This makes it work for date/time columns/